### PR TITLE
Try fixing relative-to-screen-size calculation

### PR DESCRIPTION
**Pull request by thawk**
Add an unittest for #798 , and make an attampt to fix that.
Not a perfect solution, just work.

### CODE REVIEW

File: src/impress.js [hunk 1]
```diff
@@ -335,18 +354,9 @@
...
-                maxScale: lib.util.toNumber( rootData.maxScale, defaults.maxScale ),
-                minScale: lib.util.toNumber( rootData.minScale, defaults.minScale ),
-                perspective: lib.util.toNumber( rootData.perspective, defaults.perspective ),
-                transitionDuration: lib.util.toNumber(
-                    rootData.transitionDuration, defaults.transitionDuration
-                )
-            };
+            // Initialize configuration object.
+            // Pre-init plugins may make some change, so we calculate it again.
+            config = buildConfig();
```
Comment (henrikingo) [inline src/impress.js:359]: Hmm... The way I see it, impress.getConfig() is global and any plugin can access it at any time and change the values. So this kind of arbitrarily resets it at one poínt in time, but doesn't actually protect anything in a robust way.

I see two options:

1. Make getConfig always return a new copy of the config:

    return buildConfig();

2. Just remove these lines and either expect plugins to be well behaved and not mess with the config, or alternatively consider it a feature that the config can be dynamically changed. (The latter is debatable to be fair.)

I'm ok with either so you can choose.

**Review by henrikingo — CHANGES_REQUESTED** [review]
Thanks. Nice catch.

I have just one small change request.

File: src/impress.js (hunk 1 above)
Comment (thawk) [reply src/impress.js]: I choose the second option.
After some searching, I found that no plugins modifies impress config, and even no plugins other than `rel` use that.  So it's safe to assume that the config WILL NOT be changed in pre-init stage.  If there's need for modify it, we can introduce a `setConfig()`, but we should consider that invoke order to make sure the effect it config modification.  So I think is's better NOT provide the `setConfig()` by now.

Comment (henrikingo) [reply src/impress.js]: I agree. And yes, when writing the above I already knew nobody currently modifies the config. I could have been more explicit about that.

**Pull request merged**